### ISSUE TITLE
Unicode dash not being detected (SC1100)

### ISSUE THREAD

**Issue by na4ma4** [koalaman/shellcheck#3283]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC1100
- My shellcheck version (`shellcheck --version` or "online"): 0.11.0 and online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

[[ $(curl ‑‑help || true) != "" ]] && echo "blah"

curl ‑‑help
```

#### Here's what shellcheck currently says:

```shell
$ shellcheck -x --extended-analysis=true -s bash -a --enable=all -P SCRIPTDIR -S style scripts/test.sh
```

#### Here's what I wanted or expected to see:

Something declaring the use of the incorrect hyphen (`‑` (\u2011 `NON-BREAKING HYPHEN` ) instead of `-`(\u002d `HYPHEN-MINUS`) )

#### Extra details

I can see that in the [code](https://github.com/koalaman/shellcheck/blob/d09551a4e6fa73d74f25b02f9322d0bf9fc602a5/src/ShellCheck/Parser.hs#L550C16-L550C76) it can look for these dashes, it kinda looks like that check only works for inside tests (the wiki page only shows where it is used in `[[ 3 –gt 2 ]]`), so maybe not detecting it is intended ?

I updated the example, using `echo` was pretty silly, but the check still works the same.

I did a small patch to src/Parser.hs, but my haskell is terrible, but it did work when I tested it on my example.

[0001-Modified-weirdDash-detection-to-work-for-arguments-a.patch](https://github.com/user-attachments/files/22017364/0001-Modified-weirdDash-detection-to-work-for-arguments-a.patch)
